training/src/main/scala/com/airbnb/aerosolve/training/LinearRankerUtils.scala

```diff
@@ -119,74 +123,198 @@ object LinearRankerUtils {
     sum
   }
 
-  // Makes an example pointwise while preserving the float features.
+  /**
+    * Makes an example pointwise by combining context and flatmap with feature transforms.
+    * Each observation is further flattened to [[SparseLabeledPoint]] format where features are indexed
+    * by a specified [[AdditiveModel]].
+    */
+  def makePointwiseFloatVector(examples: RDD[Example],
+                               transformer: Transformer,
+                               params: AdditiveModelTrainer.AdditiveTrainerParams,
+                               model: AdditiveModel,
+                               isTraining: Example => Boolean = _ => true,
+                               groupSize: Int = 100): RDD[SparseLabeledPoint] = {
+    val assemblerTimer = examples.sparkContext.accumulator(0L, "pointAssembler")
```

Comment:
Correct and the time spent here is actually ineligible and would be really great if we rewrite the transform altogether in the future.